domain/src/main/java/com/anytypeio/anytype/domain/search/ProfileSubscriptionManager.kt

```diff
@@ -0,0 +1,84 @@
+package com.anytypeio.anytype.domain.search
+
+import com.anytypeio.anytype.core_models.ObjectWrapper
+import com.anytypeio.anytype.core_models.Relations
+import com.anytypeio.anytype.core_models.primitives.SpaceId
+import com.anytypeio.anytype.domain.account.AwaitAccountStartManager
+import com.anytypeio.anytype.domain.base.AppCoroutineDispatchers
+import com.anytypeio.anytype.domain.config.ConfigStorage
+import com.anytypeio.anytype.domain.library.StoreSearchByIdsParams
+import com.anytypeio.anytype.domain.library.StorelessSubscriptionContainer
+import com.anytypeio.anytype.domain.subscriptions.GlobalSubscription
+import javax.inject.Inject
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.flatMapLatest
+import kotlinx.coroutines.flow.flowOn
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.mapNotNull
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+interface ProfileSubscriptionManager : GlobalSubscription {
+
+    fun observe(): Flow<ObjectWrapper.Basic>
+    fun onStart()
+    fun onStop()
+
+    class Default @Inject constructor(
+        private val scope: CoroutineScope,
+        private val configStorage: ConfigStorage,
+        private val awaitAccountStartManager: AwaitAccountStartManager,
+        private val container: StorelessSubscriptionContainer,
+        private val dispatchers: AppCoroutineDispatchers
+    ) : ProfileSubscriptionManager {
+
+        private val state = MutableStateFlow<ObjectWrapper.Basic?>(null)
```

Comment:
better to use ProfileSubscriptionManagerState

Reply:
will improve it in the next prs